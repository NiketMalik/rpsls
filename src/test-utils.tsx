import { FC, ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { THEME } from "constants/theme";

import { StoreProvider } from "context/storeContext";
import { ThemeProvider } from "@emotion/react";

const WithGlobalProviders: FC = ({ children }) => {
  return (
    <StoreProvider>
      <ThemeProvider theme={THEME}>{children}</ThemeProvider>
    </StoreProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
) => render(ui, { wrapper: WithGlobalProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
