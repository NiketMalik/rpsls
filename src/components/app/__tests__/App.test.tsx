import React from "react";
import { render, screen } from "test-utils";

import { App } from "../";

jest.mock("components/hero", () => ({ Hero: () => null }));
jest.mock("components/board", () => ({ Board: () => null }));
jest.mock("components/toolbar", () => ({ Toolbar: () => null }));

describe("App", () => {
  it("renders without error", () => {
    const { container } = render(<App />);
    const appEle = screen.getByTestId("app");
    expect(appEle).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
