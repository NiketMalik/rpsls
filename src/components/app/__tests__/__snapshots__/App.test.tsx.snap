// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without error 1`] = `
<div>
  <div
    class="p-5 py-3 py-md-5 css-cpnos0"
    data-testid="app"
  />
</div>
`;
